### PR TITLE
Remove Doorkeeper from API

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Refactor

## Description

In order to be able to completely remove Doorkeeper from our application I needed to remove it from our API. This feature was labeled as "private alpha" before so should be safe to remove. I'll try to get clarity on whether or not this needs a changelog post.

NOTE: I originally tried to remove all of Doorkeeper in a single PR (https://github.com/forem/forem/pull/15749) but that got unwieldy so I intend to break things down as seems appropriate.

## Related Tickets & Documents

https://github.com/forem/forem/issues/15748

## QA Instructions, Screenshots, Recordings

All specs should be green

### UI accessibility concerns?

No UI changes

## Added/updated tests?

- [X] Yes

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [X] I've updated the [Developer Docs](https://developers.forem.com), see https://github.com/forem/forem-docs/pull/39
- [X] I will share this change internally with the appropriate teams

### CODE REVIEW

File: spec/requests/api/v0/articles_spec.rb [hunk 1]
```diff
@@ -932,28 +897,6 @@ def put_article(**params)
         put path, params: { article: params }.to_json, headers: headers
       end
 
-      it "returns a 403 if :write_articles scope is missing (oauth)" do
```
Comment (citizen428) [inline spec/requests/api/v0/articles_spec.rb:935]: More OAuth specific specs being removed.

File: spec/requests/api/v0/listings_spec.rb [hunk 2]
```diff
@@ -430,19 +430,6 @@ def post_listing(key: api_secret.secret, **params)
         expect(listing.cached_tag_list).to eq("discuss, javascript")
       end
     end
-
-    describe "with oauth token" do
```
Comment (citizen428) [inline spec/requests/api/v0/listings_spec.rb:434]: This is no longer needed.

File: spec/requests/api/v0/readinglist_spec.rb [hunk 3]
```diff
@@ -12,33 +15,24 @@
     end
 
     context "when request is authenticated" do
-      let(:access_token) { create :doorkeeper_access_token, resource_owner: user, scopes: "public read_articles" }
       let(:user) { create(:user) }
 
-      it "works with bearer authorization" do
```
Comment (citizen428) [inline spec/requests/api/v0/readinglist_spec.rb:18]: We no longer support bearer authentication.

File: spec/requests/api/v0/webhooks_spec.rb [hunk 4]
```diff
@@ -11,38 +11,6 @@
       create(:webhook_endpoint, user: user, target_url: "https://api.example.com/webhook")
     end
 
-    context "when accessing with oauth" do
```
Comment (citizen428) [inline spec/requests/api/v0/webhooks_spec.rb:14]: This whole block is now obsolete.

File: spec/requests/api/v0/webhooks_spec.rb [hunk 5]
```diff
@@ -202,45 +170,4 @@
       expect(response).to have_http_status(:not_found)
     end
   end
-
-  describe "authorized with doorkeeper" do
```
Comment (citizen428) [inline spec/requests/api/v0/webhooks_spec.rb:206]: This block became obsolete as well.

**Review by djuber — APPROVED** [review]
Looks right - I see the https://developers.forem.com/api page does not list this as an auth method - so no documentation removal from what I can see.

Build failed because of the admin save community name (fixed in a separate PR).  I'll merge main and push to this branch (should clear the rspec failure).

**Comment by citizen428** [comment]
> I see the developers.forem.com/api page does not list this as an auth method

Yes, the other PR got merged quickly so the docs were updated before the fact which I think is ok for a feature that was labeled "private alpha" anyway.

Thanks for taking care of fixing the build error!

**Review by Ridhwana — APPROVED** [review]
This looks good to me!

**Review by jeremyf — APPROVED** [review]

**Pull request merged**